tests/system/tsystem_misc.nim

```diff
@@ -63,6 +63,16 @@ doAssert high(float64) > low(float64)
 var s = @[1]
 s.delete(0)
 
+# deleting nonexistent index should raise #16544
+block:
+  var s = @["foo", "bar"]
+  s.delete(1)
+  doAssert @["foo"] == s
+
+  var s2 = @["foo"]
+  doAssertRaises(IndexDefect):
```

Comment:
this needs a few more tests, eg: negative indexes, index 0 when input has length 0 (implementations may change)

Reply:
Added test for index 0 when input has length 0.
Negative indexes do not compile, `delete` takes a Natural. I'm not sure if it's possible to test that something doesn't compile.

Reply:
nvm, i found the `compiles` template

Reply:
`tests/system/tsystem_misc.nim` is not ideal because it's not testing in either VM nor js; most recent tests nowdays use a pattern like this: https://github.com/nim-lang/Nim/pull/17767#issuecomment-822052627 which allows ensuring we test on those backends without test duplication. How about adding creating `tests/stdlib/tsystem.nim` in similar way moving this test there (future work can move other tests there and add more tests)

Reply:
thanks- very helpful remark, done.